sonarqube-ide-visualstudio-roslyn-plugin/src/main/java/org/sonarsource/sonarlint/visualstudio/roslyn/http/JsonRequestParser.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SonarQube Ide VisualStudio Roslyn Plugin
+ * Copyright (C) 2025-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.visualstudio.roslyn.http;
+
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.sonar.api.batch.rule.ActiveRule;
+import org.sonar.api.scanner.ScannerSide;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonarsource.api.sonarlint.SonarLintSide;
+import org.sonarsource.sonarlint.visualstudio.roslyn.SqvsRoslynSensor;
+
+import java.util.Collection;
+import java.util.Map;
+
+@ScannerSide
+@SonarLintSide(lifespan = "INSTANCE")
+public class JsonRequestParser {
```

Comment:
The way I understand what this class does, it should probably be named `JsonRequestBuilder`.

Also, it may be simpler to rely on Gson's introspection-based (de)-serialization, e.g:
* Declaring a class `AnalysisRequest` with the `fileNames` and `activeRules` fields
* Using `new Gson().toJson(analysisRequest)` to write the JSON stream